### DISCUSSION TITLE
Remote usage

### DISCUSSION THREAD

**Discussion by DeanTM** [oist/optinist#544] [Q&A] [answered]
Hi,

I'm trying to find a way to use the GUI remotely, but I can't figure out how to get the GUI to find data on the remote computer.

My data is on my remote computer, "my-remote". I find that I'm able to ssh into my-remote and then start OptiNiSt there with docker. I can then port forward the OptiNiSt port 8000 to my local computer with `ssh -fN -L 8000:127.0.0.1:8000 user@my-remote`. From here I can open the OptiNiSt on my local computer.

However, when I try to **"load"** data, the data nodes only allow me to upload data from my local computer, despite the fact that OptiNiSt is running remotely. Is there a way to allow OptiNiSt to look for my data on my-remote?

I have tried copying .tif files to the default /tmp/optinist/input ([docs here](https://optinist.readthedocs.io/en/latest/utils/cui_execution.html#input-file-settings)) hoping that this would enable the image data node to find the files with **"select image"**, but it doesn't work. Perhaps files need to be loaded before OptiNiSt is able to search this default directory?

How can I enable OptiNiSt to search for files remotely?

I realise that I should be able to create the computation graph in the GUI, then download the snakemake .yml file and run it from the command line, but in doing this I don't think I can inspect intermediate results as I build the graph.

Thank you!

**Comment by ykyamane** [comment]
Hi!

>I have tried copying .tif files to the default /tmp/optinist/input....
In the case of docker version, you need to copy your file to the /tmp/optinist/input in the file system in the docker container. I think this should work for your case. Please try.

Best,
Yukako

**Reply by DeanTM** [reply]
Thank you for your response.

I tried that now, using `docker cp` to copy the files to /tmp/optinist/input in the container. I then confirmed that the files were there by copying them back, and they were.

But still the file is not available to **"select image"**. I confirmed that optinist was running correctly by rerunning it with the sample2.tiff uploaded from my local computer.

I tried copying the file with both .tif and .tiff extensions, as below:
```
dean:input$ docker cp example.tiff [container_id]:/tmp/optinist/input
Successfully copied 265kB to [container_id]:/tmp/optinist/input
dean:input$ docker cp example.tiff [container_id]:/tmp/optinist/input/example.tif
Successfully copied 265kB to [container_id]:/tmp/optinist/input/example.tif
```
then checked that the the files were there by copying them back:
```
dean:input$ docker cp [container_id]:/tmp/optinist/input ./input_copies/
Successfully copied 73.8MB to /tmp/optinist/input/input_copies/
dean:input$ ls input_copies/
example.tif   memmap__d1_60_d2_60_d3_1_order_C_frames_2000_.mmap  sample2._rig__d1_60_d2_60_d3_1_order_F_frames_2000_.mmap
example.tiff  sample2                                             sample2.tiff
```
The files are there, but only sample2.tiff is available to be loaded:

![load_image options](https://github.com/oist/optinist/assets/11351755/10efad46-cb06-4d1a-90a1-a54604311dc6)

I would appreciate any further advice. Thank you!

**Reply by ykyamane** [reply, marked as answer]
Hi Dean, 

You need to reload the browser after copying your files to /tmp/optinist/input.
Please try if you haven't.

Another think you may want to try is to mount your data directory to the Docker container.
